src/nvim/terminal.c

```diff
@@ -618,6 +621,17 @@ static int term_movecursor(VTermPos new, VTermPos old, int visible,
   return 1;
 }
 
+// Set the terminal title. Takes ownership of the title string.
+static void buf_set_term_title(buf_T *buf, String title)
```

Comment:
`title` has type `String` so that one call site can copy the C string and the other can let the String take ownership of an already-allocated C string.